modules/core/src/main/scala/gem/Observation.scala

```diff
@@ -42,22 +43,26 @@ object Observation {
 
     /** Observations are ordered by program id and index. */
     implicit val OrderId: Order[Id] =
-      Order[Program.Id].contramap[Id](_.pid)   |+|
+      Order[Program.Id].contramap[Id](_.pid)   whenEqual
       Order[Int]       .contramap[Id](_.index)
 
     implicit val OrderingId: scala.math.Ordering[Id] =
-      OrderId.toScalaOrdering
+      OrderId.toOrdering
 
     implicit val showId: Show[Id] =
-      Show.showA
+      Show.fromToString
 
   }
 
   /** Observation is a bitraversable functor. */
   implicit val ObservationBitraverse: Bitraverse[Observation] =
     new Bitraverse[Observation] {
-      def bitraverseImpl[G[_]: Applicative, A, B, C, D](fab: Observation[A,B])(f: A => G[C], g: B => G[D]): G[Observation[C,D]] =
-        (f(fab.staticConfig) |@| fab.steps.traverse(g))((c, d) => fab.copy(staticConfig = c, steps = d))
+      def bifoldLeft[A, B, C](fab: Observation[A,B], c: C)(f: (C, A) => C, g: (C, B) => C): C =
+        fab.steps.foldLeft(f(c, fab.staticConfig))(g)
+      def bifoldRight[A, B, C](fab: Observation[A,B], c: Eval[C])(f: (A, Eval[C]) => Eval[C], g: (B, Eval[C]) => Eval[C]): Eval[C] =
+        fab.steps.foldRight(f(fab.staticConfig, c))(g)
+      def bitraverse[G[_]: Applicative, A, B, C, D](fab: Observation[A, B])(f: A => G[C], g: B => G[D]): G[Observation[C,D]] =
+        (f(fab.staticConfig), fab.steps.traverse(g)).mapN((c, d) => fab.copy(staticConfig = c, steps = d))
```

Comment:
Yeah, cats doesn't provide default implementations of typeclass methods when the default is inefficient, as with `foldRight` in terms of `traverse` (which is implemented by traversing with `State[List[A], ?]`). I argued against this but lost. In practice it doesn't come up a lot.